### PR TITLE
Feature/cvsb 10372

### PR DESCRIPTION
**Pull request by DeepikaSinghD**

### CODE REVIEW

File: src/main/resources/loader/develop/test-results_car_post_payload_10372.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      "additionalNotesRecorded":"zdfvc",
+      "defects":[
+
+      ],
+      "customDefects":[
+
+      ]
+    }
+  ]
+}
```
Comment (dpanzaru-deloitte) [inline src/main/resources/loader/develop/test-results_car_post_payload_10372.json:64]: New line here at the end of the file

File: src/main/resources/loader/develop/test-results_car_put_payload_10372.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            "referenceNumber":"string",
+            "defectName":"string",
+            "defectNotes":"string"
+          }
+
+        ]
+      }
+    ]
+  }
+}
```
Comment (dpanzaru-deloitte) [inline src/main/resources/loader/develop/test-results_car_put_payload_10372.json:65]: Same as above

File: src/main/resources/loader/develop/test-results_motorcycle_post_payload_10372.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      "additionalNotesRecorded":"zdfvc",
+      "defects":[
+
+      ],
+      "customDefects":[
+
+      ]
+    }
+  ]
+}
```
Comment (dpanzaru-deloitte) [inline src/main/resources/loader/develop/test-results_motorcycle_post_payload_10372.json:61]: Same as above

File: src/main/resources/loader/develop/test-results_motorcycle_put_payload_10372.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            "referenceNumber":"string",
+            "defectName":"string",
+            "defectNotes":"string"
+          }
+
+        ]
+      }
+    ]
+  }
+}
```
Comment (dpanzaru-deloitte) [inline src/main/resources/loader/develop/test-results_motorcycle_put_payload_10372.json:63]: Same as above

File: src/main/resources/loader/develop/test-results_psv_post_payload_10372.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      "additionalNotesRecorded":"zdfvc",
+      "defects":[
+
+      ],
+      "customDefects":[
+
+      ]
+    }
+  ]
+}
```
Comment (dpanzaru-deloitte) [inline src/main/resources/loader/develop/test-results_psv_post_payload_10372.json:66]: same as above

File: src/main/resources/loader/develop/test-results_psv_put_payload_10372.json [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+          {
+            "referenceNumber":"string",
+            "defectName":"string",
+            "defectNotes":"string"
+          }
+        ]
+      }
+    ]
+  }
+}
```
Comment (dpanzaru-deloitte) [inline src/main/resources/loader/develop/test-results_psv_put_payload_10372.json:70]: Same as above

File: src/test/java/testresults/PutTestResultsFieldRestrictionsForIVA.java [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+                alterationPutTestEndTimestamp,
+                alterationPutTestTypeStartTimestamp,
+                alterationPutTestTypeEndTimestamp,
+                restriction
+        ));
+
+        testResultsSteps.putTestResultsWithAlterations(randomSystemNumber, putRequestBody, alterationsPutTestResults);
+        testResultsSteps.statusCodeShouldBe(HttpStatus.SC_BAD_REQUEST);
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsFieldRestrictionsForIVA.java:192]: Empty new line here

File: src/test/java/testresults/PutTestResultsFieldRestrictionsForMSVA.java [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+                alterationPutTestEndTimestamp,
+                alterationPutTestTypeStartTimestamp,
+                alterationPutTestTypeEndTimestamp,
+                restriction
+        ));
+
+        testResultsSteps.putTestResultsWithAlterations(randomSystemNumber, putRequestBody, alterationsPutTestResults);
+        testResultsSteps.statusCodeShouldBe(HttpStatus.SC_BAD_REQUEST);
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsFieldRestrictionsForMSVA.java:193]: Same as above

File: src/test/java/testresults/PutTestResultsForCOIFWithAnnual.java [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+        testResultsSteps.valueForFieldInPathShouldBe("createdByName","catalin");
+        testResultsSteps.valueForFieldInPathShouldBe("createdById","123243424-234234245");
+        testResultsSteps.valueForFieldInPathShouldContains("createdAt",new SimpleDateFormat("yyyy-MM-dd").format(date));
+
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].testVersion","archived");
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].lastUpdatedByName","catalin");
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].lastUpdatedById","123243424-234234245");
+        testResultsSteps.valueForFieldInPathShouldContains("testHistory[0].lastUpdatedAt",new SimpleDateFormat("yyyy-MM-dd").format(date));
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsForCOIFWithAnnual.java:179]: Same as above

File: src/test/java/testresults/PutTestResultsForCOIFWithoutAnnual.java [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+        testResultsSteps.valueForFieldInPathShouldBe("createdById","123243424-234234245");
+        testResultsSteps.valueForFieldInPathShouldContains("createdAt",new SimpleDateFormat("yyyy-MM-dd").format(date));
+
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].testVersion","archived");
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].lastUpdatedByName","catalin");
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].lastUpdatedById","123243424-234234245");
+        testResultsSteps.valueForFieldInPathShouldContains("testHistory[0].lastUpdatedAt",new SimpleDateFormat("yyyy-MM-dd").format(date));
+
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsForCOIFWithoutAnnual.java:200]: Same as above

File: src/test/java/testresults/PutTestResultsForIVA.java [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+                alterationPutTestEndTimestamp,
+                alterationPutTestTypeStartTimestamp,
+                alterationPutTestTypeEndTimestamp,
+                restriction
+        ));
+
+        testResultsSteps.putTestResultsWithAlterations(randomSystemNumber, putRequestBody, alterationsPutTestResults);
+        testResultsSteps.statusCodeShouldBe(HttpStatus.SC_OK);
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsForIVA.java:174]: Same as above

File: src/test/java/testresults/PutTestResultsForMSVA.java [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+        testResultsSteps.valueForFieldInPathShouldBe("createdByName","catalin");
+        testResultsSteps.valueForFieldInPathShouldBe("createdById","123243424-234234245");
+        testResultsSteps.valueForFieldInPathShouldContains("createdAt",new SimpleDateFormat("yyyy-MM-dd").format(date));
+
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].testVersion","archived");
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].lastUpdatedByName","catalin");
+        testResultsSteps.valueForFieldInPathShouldBe("testHistory[0].lastUpdatedById","123243424-234234245");
+        testResultsSteps.valueForFieldInPathShouldContains("testHistory[0].lastUpdatedAt",new SimpleDateFormat("yyyy-MM-dd").format(date));
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsForMSVA.java:186]: Same as above

File: src/test/java/testresults/PutTestResultsWithoutMandatoryForCOIFWithAnnual.java [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+                alterationPutTestEndTimestamp,
+                alterationPutTestTypeStartTimestamp,
+                alterationPutTestTypeEndTimestamp,
+                restriction
+        ));
+
+        testResultsSteps.putTestResultsWithAlterations(randomSystemNumber, putRequestBody, alterationsPutTestResults);
+        testResultsSteps.statusCodeShouldBe(HttpStatus.SC_BAD_REQUEST);
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsWithoutMandatoryForCOIFWithAnnual.java:189]: Same as above

File: src/test/java/testresults/PutTestResultsWithoutMandatoryForCOIFWithoutAnnual.java [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+                alterationPutTestEndTimestamp,
+                alterationPutTestTypeStartTimestamp,
+                alterationPutTestTypeEndTimestamp,
+                restriction
+        ));
+
+        testResultsSteps.putTestResultsWithAlterations(randomSystemNumber, putRequestBody, alterationsPutTestResults);
+        testResultsSteps.statusCodeShouldBe(HttpStatus.SC_BAD_REQUEST);
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsWithoutMandatoryForCOIFWithoutAnnual.java:188]: Same as above

File: src/test/java/testresults/PutTestResultsWithoutMandatoryForIVA.java [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+                alterationPutTestEndTimestamp,
+                alterationPutTestTypeStartTimestamp,
+                alterationPutTestTypeEndTimestamp,
+                restriction
+        ));
+
+        testResultsSteps.putTestResultsWithAlterations(randomSystemNumber, putRequestBody, alterationsPutTestResults);
+        testResultsSteps.statusCodeShouldBe(HttpStatus.SC_BAD_REQUEST);
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsWithoutMandatoryForIVA.java:183]: Same as above

File: src/test/java/testresults/PutTestResultsWithoutMandatoryForMSVA.java [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+                alterationPutTestEndTimestamp,
+                alterationPutTestTypeStartTimestamp,
+                alterationPutTestTypeEndTimestamp,
+                restriction
+        ));
+
+        testResultsSteps.putTestResultsWithAlterations(randomSystemNumber, putRequestBody, alterationsPutTestResults);
+        testResultsSteps.statusCodeShouldBe(HttpStatus.SC_BAD_REQUEST);
+    }
+}
```
Comment (dpanzaru-deloitte) [inline src/test/java/testresults/PutTestResultsWithoutMandatoryForMSVA.java:181]: Same as above

File: src/main/resources/loader/develop/test-results_car_post_payload_10372.json (hunk 1 above)
Comment (DeepikaSinghD) [reply src/main/resources/loader/develop/test-results_car_post_payload_10372.json]: Changes done.Pls review

**Pull request merged**